components/authz-service/server/v2/rules_test.go

```diff
@@ -569,6 +569,105 @@ func TestListRules(t *testing.T) {
 	}
 }
 
+func TestListRulesForProject(t *testing.T) {
+	ctx := context.Background()
+	cl, store, _, _ := setupRules(t)
+
+	storageConditions1 := []storage.Condition{
+		{
+			Type:      storage.Node,
+			Attribute: storage.Organization,
+			Operator:  storage.MemberOf,
+			Value:     []string{"opscode", "chef"},
+		},
+	}
+	apiConditions2 := []*api.Condition{
+		{
+			Type:     api.ProjectRuleConditionTypes_CHEF_ORGS,
+			Values:   []string{"chef"},
+			Operator: api.ProjectRuleConditionOperators_EQUALS,
+		},
+	}
+	storageConditions2 := []storage.Condition{
+		{
+			Type:      storage.Event,
+			Attribute: storage.Organization,
+			Operator:  storage.Equals,
+			Value:     []string{"chef"},
+		},
+	}
+	apiConditions3 := []*api.Condition{
+		{
+			Type:     api.ProjectRuleConditionTypes_CHEF_ORGS,
+			Values:   []string{"other", "org"},
+			Operator: api.ProjectRuleConditionOperators_MEMBER_OF,
+		},
+	}
+	storageConditions3 := []storage.Condition{
+		{
+			Type:      storage.Event,
+			Attribute: storage.Organization,
+			Operator:  storage.MemberOf,
+			Value:     []string{"other", "org"},
+		},
+	}
+
+	cases := []struct {
+		desc string
+		f    func(*testing.T)
+	}{
+		{"if project does not exist, returns empty list", func(t *testing.T) {
```

Comment:
Shouldn't this be a "not found" ?

Reply:
Wasn't sure. You are listing things so I didn't know if we wanted to return a 404 there.

Reply:
I'd need to add another query to the DB code to do that (aka check if the project exist), can do that if we think that's the right thing.

Reply:
Perhaps leave it as is for now, and let's see how things fall out as we progress.